### DISCUSSION TITLE
Dynamic Secure Pattern

### DISCUSSION THREAD

**Discussion by gregmcguffey** [uNmAnNeR/imaskjs#883] [Q&A]
Is there a way to dynamically use a secure pattern? Use case is for SSN entry. Have '*' as display character normally (so it's secure), but allow user to see actual value if they click on a "show" button.  I tried to dynamically set the displayChar of the definitions prop but it did nothing.

**Comment by uNmAnNeR** [comment]
should work if you update it using `updateOptions`

**Comment by gregmcguffey** [comment]
I'm not see how that would be used. I'm using the React IMaskInput component.

**Reply by uNmAnNeR** [reply]
I assume that you update the property inside an object and react does not update the view. If you want something more concrete please create a small example to run somewhere e.g. codesandbox.
